src/metabase/lib/util.cljc

```diff
@@ -56,11 +56,14 @@
   (m/update-existing-in
     stage
     location
-    #(->> (for [clause %]
-            (if (= (clause-uuid clause) (clause-uuid target-clause))
-              new-clause
-              clause))
-          vec)))
+    (fn [clause-or-clauses]
+      (if (= :expressions (first location))
+        new-clause
```

Comment:
The asymetry is because expressions are a map rather than a list of clauses. `location` will be `[:expressions "my-expr"]` so only that particular clause gets replaced.